### PR TITLE
Update CI

### LINKED ISSUE #65: CI fails for Python 3.9
## Expected behavior ##

All CI checks pass.

## Actual behavior ##

CI fails for Python 3.9 because it cannot install an appropriate development version of MDAnalysis. This is due to https://github.com/MDAnalysis/cookiecutter-mdakit/issues/110 .

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #65 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
I'm leaving packaging alone for now and will fix #67 separately

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ The rules for this file:
 
 ### Changed
 - Update CI and wheel deployment to include macos-arm64 and py3.12 (PR #63)
+- Minimum Python version has been raised to 3.10 (PR #66)
+- Minimum MDAnalysis version has been raisesd to 2.1.0 (PR #66)
```
Comment (orbeckst) [inline CHANGELOG.md:36]:
```suggestion
- Minimum MDAnalysis version has been raised to 2.1.0 (PR #66)
```

**Review by orbeckst — APPROVED** [review]
This PR fixes the issue for the short-term at the cost of removing some automation.

Perhaps add comments to the CI files as a note which lines will need to be changed with a reference to pertinent issue numbers? Otherwise lgtm as it allows us to move forward.

**Comment by IAlibay** [comment]
> Perhaps add comments to the CI files as a note which lines will need to be changed with a reference to pertinent issue numbers? Otherwise lgtm as it allows us to move forward.

Done!

**Pull request merged**